src/jobs/go_build_test.yml

```
@@ -0,0 +1,39 @@
+description: >
+  Job for building and testing go app
+
+parameters:
+  executor:
+    type: executor
+    deafault:
+      name: go/default
+      tag: '1.17'
+  work_dir:
+    type: string
+    default: .
+  additional_tests:
+    description: 'Steps with additional tests e.g. tests on common'
+    type: steps
+    default: [ ]
+  go_mod_cache_path:
+    default: /home/circleci/go/pkg/mod
+    description: 'Path to cache. Default path works for default executor (cimg/go)'
+    type: string
+
+executor: << parameters.executor >>
+
+steps:
+  - checkout
+  - restore_cache:
+      keys:
+        - 'go-mod-{{ arch }}-{{ checksum "<< parameters.work_dir >>/go.sum"  }}'
```

for java we did something like this to attach caches from previous builds in case this exact file (go.sum) was not built before - not sure whether such a fallback makes sense in case of go though. do you know?
```suggestion
  - restore_cache:
      keys:
        - 'go-mod-{{ arch }}-{{ checksum "<< parameters.work_dir >>/go.sum"  }}'
        - 'go-mod-{{ arch }}-'
```